packages/ethers/src/index.ts

```diff
@@ -260,6 +259,19 @@ export class TurnkeySigner extends ethers.Signer implements TypedDataSigner {
   }
 
   _signTypedData = this.signTypedData.bind(this);
+
+  override async sendTransaction(tx: TransactionRequest): Promise<TransactionResponse> {
```

Comment:
In upgrading from v5 to v6, I had to override `sendTransaction` due to a nuanced type issue between `AbstractSigner` and `Transaction`:

The core of the issue is that `sendTransaction` generates a `Transaction` instance via `Transaction.from()`, which is then passed to `signTransaction`. 
Although `signTransaction` is designed to accept a `TransactionRequest`, the structural similarities between `TransactionRequest` and `Transaction` mean TypeScript doesn't identify this as an error, despite a crucial difference: `Transaction`'s properties are private. 
This distinction becomes problematic when `resolveProperties` is invoked on the `TransactionRequest` within `signTransaction`, leading to an empty object because `Object.keys()` cannot access `Transaction`'s private properties. 
Importantly, `TransactionRequest` may contain properties as promises, necessitating the use of `resolveProperties` to ensure proper handling. 
This subtlety in how `Transaction` instances and `TransactionRequest` objects are treated, especially regarding promise properties and the need for `resolveProperties`, indicates a potential area for improvement in ethers library's type system.